app/src/main/java/com/kickstarter/ui/activities/CommentsActivity.kt

```diff
@@ -18,7 +20,24 @@ class CommentsActivity :
     BaseActivity<CommentsViewModel.ViewModel>(),
     CommentsAdapter.Delegate {
     private lateinit var binding: ActivityCommentsLayoutBinding
-    private val adapter = CommentsAdapter(this)
+    private val adapter = CommentsAdapter(
+        this,
+        object : DiffUtil.ItemCallback<Any>() {
+            override fun areItemsTheSame(oldItem: Any, newItem: Any): Boolean {
+                return threadsAreTheSame(oldItem, newItem)
+            }
+
+            override fun areContentsTheSame(oldItem: Any, newItem: Any): Boolean {
+                return threadsAreTheSame(oldItem, newItem)
+            }
+
+            private fun threadsAreTheSame(oldItem: Any, newItem: Any): Boolean {
+                val oldThread = oldItem as Comment
+                val newThread = newItem as Comment
+                return oldThread.id() == newThread.id()
```

Comment:
if the equals operator if properly implemented by doing `oldObj == newObj ` should be enough, so we can make the DiffUtil generic

Reply:
Done